test/unit/deprecated_references_test.rb

```diff
@@ -25,6 +25,79 @@ class DeprecatedReferencesTest < Minitest::Test
       )
     end
 
+    test "#stale_violations? returns true if deprecated references exist but no violations can be found in code" do
```

Comment:
These four tests are (very nicely! 👏 ) unit testing just `DeprecatedReferences#stale_violations?`, which doesn’t depend on any of the other machinery being added simultaneously in `DetectStaleDeprecatedReferences` and `Cli`. Would you consider pulling this out into a separate commit, i.e. first implement & test `DeprecatedReferences#stale_violations?`, then add everything else afterwards? Having these in separate commits would make it more obvious which changes go together.